esmf_regrid/tests/unit/schemes/test_regrid_rectilinear_to_rectilinear.py

```diff
@@ -0,0 +1,150 @@
+"""Unit tests for :func:`esmf_regrid.schemes.regrid_rectilinear_to_rectilinear`."""
+
+from iris.coord_systems import RotatedGeogCS
+from iris.coords import AuxCoord, DimCoord
+from iris.cube import Cube
+import numpy as np
+from numpy import ma
+
+from esmf_regrid.schemes import regrid_rectilinear_to_rectilinear
+from esmf_regrid.tests.unit.schemes.test__cube_to_GridInfo import _grid_cube
+
+
+def test_rotated_regridding():
+    """
+    Test for :func:`esmf_regrid.schemes.regrid_rectilinear_to_rectilinear`.
+
+    Test the regriding of a rotated pole coordinate system. The test is
+    designed to that it should be possible to verify the result by
+    inspection.
+    """
+    src_coord_system = RotatedGeogCS(0, 90, 90)
+    tgt_coord_system = None
+
+    n_lons = 4
+    n_lats = 4
+    lon_bounds = (-180, 180)
+    lat_bounds = (-90, 90)
+
+    src = _grid_cube(
+        n_lons,
+        n_lats,
+        lon_bounds,
+        lat_bounds,
+        circular=True,
+        coord_system=src_coord_system,
+    )
+    tgt = _grid_cube(
+        n_lons,
+        n_lats,
+        lon_bounds,
+        lat_bounds,
+        circular=True,
+        coord_system=tgt_coord_system,
+    )
+    src_data = np.arange(n_lons * n_lats).reshape([n_lats, n_lons])
+    # src_mask = np.empty([n_lats, n_lons])
+    # src_mask[:] = np.array([1, 0, 0, 1])[:, np.newaxis]
+    src_mask = np.array([[1, 1, 1, 1], [0, 0, 0, 0], [0, 0, 0, 0], [1, 1, 1, 1]])
+    src_data = ma.array(src_data, mask=src_mask)
+    src.data = src_data
+
+    no_mdtol_result = regrid_rectilinear_to_rectilinear(src, tgt)
+
+    full_mdtol_result = regrid_rectilinear_to_rectilinear(src, tgt, mdtol=1)
+
+    expected_data = np.array(
+        [[5, 4, 8, 9], [5, 4, 8, 9], [6, 7, 11, 10], [6, 7, 11, 10]]
+    )
+    expected_mask = np.array([[0, 0, 0, 0], [1, 1, 1, 1], [1, 1, 1, 1], [0, 0, 0, 0]])
+    no_mdtol_expected_data = ma.array(expected_data, mask=expected_mask)
+
+    # Lenient check for data.
+    assert np.allclose(no_mdtol_expected_data, no_mdtol_result.data)
+    assert np.allclose(expected_data, full_mdtol_result.data)
+
+
+def test_extra_dims():
+    """
+    Test for :func:`esmf_regrid.schemes.regrid_rectilinear_to_rectilinear`.
+
+    Tests the handling of extra dimensions and metadata.
```

Comment:
Maybe expand on this a little bit more.  i.e. what are we expecting the handling to do?  Pass through all metadata and extra dims?  Expect the ordering of dims to be the same or different?